### ISSUE TITLE
[Bazel CI] Bazel fails to build app with App Intents due to AppIntentsMetadataProcessor failure


### ISSUE THREAD

**Issue by sgowroji** [bazelbuild/rules_apple#2488]
CI: https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/3973#01909085-4949-49c3-b0ab-9bacc86a8657

 Platform: MacOS

 Logs: 
```
ERROR: [0m/Users/buildkite/builds/bk-macos-arm64-h2e1/bazel-org-repo-root/rules_apple/test/starlark_tests/targets_under_test/ios/BUILD:4462:16: AppIntentsMetadataProcessor test/starlark_tests/targets_under_test/ios/app_with_app_intents-intermediates/Metadata.appintents failed: (Exit 255): bash failed: error executing AppIntentsMetadataProcessor command (from target //test/starlark_tests/targets_under_test/ios:app_with_app_intents) 
error:* ]]; then
ERROR: [0mBuild did NOT complete successfully
Error: The command exited with status 1
error: exit status 1
``` 
 Culprit: 

 Steps:
  ``` 
 git clone https://github.com/bazelbuild/rules_apple
git reset 42fb193ca0e5ad0b95ccad47c9cc663ae8c86f5b  --hard
export USE_BAZEL_VERSION=8a244ccb5bd484c45bec4446211bf95c32683ac9
bazel build  tools/... test/... 
 ``` 
 CC Greenteam @comius

**Comment by keith** [comment]
should be fixed by https://github.com/bazelbuild/rules_apple/pull/2418

**Comment by BalestraPatrick** [comment]
It feels like the CI machines should still have Xcode 15.1 installed instead of immediately only using Xcode 15.4 which is causing these failures. There's another breakage in rules_apple that requires to be fixes in order to support Xcode 15.3+, so we should find a way to only use machines with Xcode 15.1 installed (or make sure Xcode 15.1 is installed on all macOS machines).

**Comment by luispadron** [comment]
This should be fixed after #2418 was merged
